modules/library/src/main/scala/zio/elasticsearch/ElasticAggregation.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2022 LambdaWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.elasticsearch
+
+import zio.elasticsearch.ElasticAggregation.MultipleAggregations
+import zio.elasticsearch.ElasticPrimitive._
+import zio.json.ast.Json
+import zio.json.ast.Json.Obj
+
+sealed trait ElasticAggregation { self =>
+  def paramsToJson: Json
+
+  final def toJson: Json =
+    Obj("aggs" -> paramsToJson)
+}
+
+sealed trait HasSubAgg[A <: HasSubAgg[A]] {
+  def withSubAgg(subAgg: SingleElasticAggregation): A
+}
+
+sealed trait HasOtherAgg {
+  def withAgg(agg: SingleElasticAggregation): MultipleAggregations
+}
+
+sealed trait SingleElasticAggregation extends ElasticAggregation
+
+object ElasticAggregation {
+  def multipleAggregations: Multiple =
+    Multiple.empty
```

Comment:
Would it make sense to drop this method and instead construct `Multiple` with a method taking var args, like we've done with `BoolQuery`?